Modix/Handlers/InvitePurgeHandler.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Discord;
+using Discord.WebSocket;
+using Modix.Services.BehaviourConfiguration;
+using Serilog;
+
+namespace Modix.Handlers
+{
+    public class InviteLinkHandler
+    {
+        private readonly IBehaviourConfiguration _botConfiguration;
+
+        private const string InviteLinkPattern = @"(https?:\/\/)?(www\.)?(discord\.(gg|io|me|li)|discordapp\.com\/invite)\/.+[a-z]";
+
+        public InviteLinkHandler(IBehaviourConfiguration botConfiguration)
+        {
+            _botConfiguration = botConfiguration;
+        }
+
+        /// <summary>
+        /// Will scan a message for any matchable Discord invite links
+        /// that are not part of the context's current Guild.
+        /// </summary>
+        /// <param name="message"></param>
+        /// <returns>False if no Invite link was found, true if an invite link was found and the message was purged</returns>
+        public async Task<bool> PurgeInviteLink(IMessage message)
+        {
+            if (!(message.Channel is SocketGuildChannel channel))
+                return false;
+
+            if (!(message.Author is SocketGuildUser user))
+                return false;
+
+            if (!_botConfiguration.InvitePurgeBehaviour.IsEnabled)
+                return false;
```

Comment:
These three conditions can be simplified into a single `if` statement.

Reply:
I've intentionally left them to make it more readable, I think it works out better separated?